api/src/public/login/script.js

```diff
@@ -152,8 +154,13 @@ const parseTranslations = function() {
   return JSON.parse(decodeURIComponent(raw));
 };
 
+const checkUsername = function() {
+  const usernameQueryParam = urlParams.get('username');
+  const usernameEntered    = document.getElementById('user').value.toLowerCase().trim();
+  return usernameQueryParam === usernameEntered;
+};
+
 const getRedirectUrl = function() {
-  const urlParams = new URLSearchParams(window.location.search);
```

Comment:
I think it'd be cleaner to update the `getRedirectUrl()` function to check the username and return the right thing. That way `urlParams` doesn't need to be a global constant and it can be evaluated late. For example...

```js
const getRedirectUrl = function() {
  const urlParams = new URLSearchParams(window.location.search);
  const usernameQueryParam = urlParams.get('username');
  const usernameEntered = document.getElementById('user').value.toLowerCase().trim();
  if (usernameQueryParam === usernameEntered) {
      return urlParams.get('redirect');
  }
};
```

Reply:
Hi @garethbowen! Thanks for reviewing this PR.
I also created `checkUsername` function in order not to modify `getRedirectUrl` function because it is also used in `checkSession` function and I did not know if this could cause any regression in the latter.
Please tell me if you think I can update `getRedirectUrl` without causing any problem or where I can check it works as expected if I will update it.
Thanks.

Reply:
Thanks for pointing that out! Yes I think we should update the `checkSession` function too.

Reply:
I updated the `getRedirectUrl` function as you suggested. Thanks for the help.